pkg/proxy/iptables/proxier.go

```diff
@@ -1033,7 +1033,7 @@ func (proxier *Proxier) syncProxyRules() {
 			// If the "external" IP happens to be an IP that is local to this
 			// machine, hold the local port open so no other process can open it
 			// (because the socket might open but it would never work).
-			if localAddrSet.Len() > 0 && (svcInfo.Protocol() != v1.ProtocolSCTP) && localAddrSet.Has(net.ParseIP(externalIP)) {
+			if (svcInfo.Protocol() != v1.ProtocolSCTP) && localAddrSet.Has(net.ParseIP(externalIP)) {
```

Comment:
if `externalIP` is invalid `net.ParseIP(externalIP)` will be nil  and this will panic if there is no `localAddrSet.Len() > 0`

I've tested it with 
```
func TestIPNets(t *testing.T) {
	s := IPNetSet{}
	if s.Has(nil) {
		fmt.Println("ok")
	}
```
and panics, however with 

```
	s := IPNetSet{}
	if len(s) > 0 && s.Has(nil) {
		fmt.Println("ok")
	}
````
it does not panic

Reply:
That's only because we are using an empty set and the `Has()` method is never called. If there was an invalid externalIP and we still had len > 0 local addresses it would still panic.